src/main/java/com/atlassian/db/replica/internal/ReplicaStatement.java

```diff
@@ -24,7 +24,7 @@
     private final List<StatementOperation<Statement>> batches = new ArrayList<>();
     private final ReplicaConsistency consistency;
     private final DatabaseCall databaseCall;
-    private boolean isWriteOperation = true;
```

Comment:
@wyrzyk 
> The change can't be rolled-out without disabling DualConnection. I created PRs for two propositions on how to roll out changes safely (#52 and #40). We can choose one of them or a different approach, but we shouldn't release it without any safe-rollout mechanism.

Oh man! This is scope explosion. Though I understand your intentions. It's just those changes aren't merged yet.
Just to double check,
what's the risk of rolling out `DualConnection` with this change from yours perspective? From my point the only risky is removing `isWriteOperation`  internal field, which is proven by unit tests, what give me confidence on compile lvl to go 👍 

Maybe I can extract this change and then merge It once https://github.com/atlassian-labs/db-replica/pull/52 will be merged?

Reply:
> what's the risk of rolling out DualConnection with this change from yours perspective? 

We need to use FF for every rollout (deployment may run outside of our working hours and we are expected to observe the rollout). We also should stop to use global kill-switch. So far I have two propositions about how to solve this problem, I'm also open to other solutions.

Reply:
Fair enough..,
So we need to prioritize (#52 and #40) 🙏 
Let me jump on reviewing (#52 and #40) to push it further as it blocks this story 🚫

Reply:
* Separate branch for refactor https://github.com/atlassian-labs/db-replica/pull/54
* This branch will left only with changes required for adding isWrite API